src/admin.tsx

```diff
@@ -54,6 +56,66 @@ export default function AdminPage() {
     return unsubscribe;
   }, [db, setData]);
 
+  // Snackbar section
+  useEffect(() => {
+    if (snackPack.length && !snackBarMessage) {
+      // Set a new snack when we don't have an active one
+      setSnackBarMessage(snackPack[0]);
+      setSnackPack((prev) => prev.slice(1));
+      setOpen(true);
+    } else if (snackPack.length && snackBarMessage) {
+      // Close an active snack when a new one is added
+      setOpen(false);
+    }
+  }, [snackPack, snackBarMessage]);
+
+  const handleSnackBarExited = () => {
+    setSnackBarMessage("");
+    setOpen(false);
+  }
+
+  const handleAddSnackBarMessage = (message: string) => {
+    setSnackPack((prev) => [...prev, message]);
+  }
+
+  const handleCloseSnackBar = (event: any, reason: any) => {
+    if (reason === 'clickaway') {
+      return;
+    }
+    setSnackBarMessage("");
+    setOpen(false);
+  };
+
+  // Expand / Collapse Tree Nodes 
+  const handleExpandedItemsChange = (
+    event: React.SyntheticEvent,
+    itemIds: string[],
+  ) => {
+    setExpandedItems(itemIds);
+  };
+
+  const handleExpandClick = () => {
+    if (!data) {
+      return;
+    }
+
+    const videosBySubCat = groupVideosBySubcategory(data);
+    const expandedItems: string[] = [];
+
+    Object.entries(videosBySubCat).forEach(([documentID, subcategoryGrouping]) => {
+      expandedItems.push(documentID);
+      Object.keys(subcategoryGrouping.videosBySubcategory).forEach((subcategory) => {
+        expandedItems.push(documentID + subcategory);
+      })
+    })
```

Comment:
So this list is generated _every_ time we press the button. Can we find a way to only generate this once? Maybe a new state variable which we populate in the existing useEffect after we get data from firebase?